v2/mongodb-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/mongodb/templates/MongoDbToBigQuery.java

```diff
@@ -65,30 +71,54 @@ public void processElement(ProcessContext context) {
     }
   }
 
-  public static void main(String[] args) {
+  public static void main(String[] args)
+      throws ScriptException, IOException, NoSuchMethodException {
     Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
 
     BigQueryIOUtils.validateBQStorageApiOptionsBatch(options);
 
     run(options);
   }
 
-  public static boolean run(Options options) {
+  public static boolean run(Options options)
+      throws ScriptException, IOException, NoSuchMethodException {
     Pipeline pipeline = Pipeline.create(options);
-    TableSchema bigquerySchema =
-        MongoDbUtils.getTableFieldSchema(
-            options.getMongoDbUri(),
-            options.getDatabase(),
-            options.getCollection(),
-            options.getUserOption());
     String userOption = options.getUserOption();
-    pipeline
-        .apply(
-            "Read Documents",
-            MongoDbIO.read()
-                .withUri(options.getMongoDbUri())
-                .withDatabase(options.getDatabase())
-                .withCollection(options.getCollection()))
+
+    TableSchema bigquerySchema;
+
+    if (options.getUserOption().equals("UDF")) {
+      bigquerySchema =
+          MongoDbUtils.getTableFieldSchemaForUDF(
+              options.getMongoDbUri(),
+              options.getDatabase(),
+              options.getCollection(),
+              options.getJavascriptDocumentTransformGcsPath(),
+              options.getJavascriptDocumentTransformFunctionName());
+    } else {
+      bigquerySchema =
+          MongoDbUtils.getTableFieldSchema(
+              options.getMongoDbUri(),
+              options.getDatabase(),
+              options.getCollection(),
+              options.getUserOption());
+    }
+    PCollection<Document> lines =
```

Comment:
nit: Storing in a variable seems redundant if nothing is done with `lines`.

Reply:
Updated.